gramps/plugins/test/test_exports.py

```diff
@@ -0,0 +1,218 @@
+            retval = False
+        elif token == "DATE" and gedfilt.prev[gedfilt.indx-2][0] == "NAME":
+            # we must have a header with file date
+            retval = False
+        elif token == "TIME" and gedfilt.prev[gedfilt.indx-2][0] == "DATE":
+            # probably have a header with file time
+            retval = False
+        elif token == "FILE" and line.endswith('.ged\n'):
+            # probably have a header with file name
+            retval = False
+    else:   # this is an addition
+        if token == "VERS" and gedfilt.prev[gedfilt.indx-1][0] == "VERS":
+            # we must have a header with Gramps version
+            retval = False
+        elif token == "DATE" and (gedfilt.prev[gedfilt.indx-2][0] == "NAME" or
+                                  gedfilt.prev[gedfilt.indx-3][0] == "NAME"):
+            # we must have a header with file date
+            retval = False
+        elif token == "TIME" and (gedfilt.prev[gedfilt.indx-2][0] == "DATE" or
+                                  gedfilt.prev[gedfilt.indx-3][0] == "DATE"):
+            # probably have a header with file time
+            retval = False
+        elif token == "FILE" and line.endswith('.ged\n'):
+            # probably have a header with file name
+            retval = False
+    return retval
+
+
+def vcffilt(line):
+    """ A filter for VCard files.
+    This implements a filter that allows some differences to be ignored.
+    The differences are not functional, but are related to changes in Gramps
+    version.
+    """
+    return False if line.startswith('PRODID:') else True
+
+
+class ExportControl(unittest.TestCase):
+    """ These tests compare various exported files with expected files,
+    based on the matching '.gramps' test file as a source.
+    As more types of exports are tested, we will need to provide some
+    filters for the differences; some types of exports have Gramps versions,
+    export dates, file names etc. that don't count as differences.
+    """
+    def setUp(self):
+        self.tearDown()  # removes it if it existed
+
+#        out, err = self.call("-C", TREE_NAME,
+#                            "--import", example)
+
+    def tearDown(self):
+        call("-y -q", "--remove", TREE_NAME)
+
+    #===========================================================================
+    # def test_csv(self):
+    #     """ Run a csv export test """
+    #     set_format(0)   # Use ISO date for test
+    #     src_file = 'exp_sample_csv.gramps'
+    #     tst_file = 'exp_sample_csv.csv'
+    #     msg = do_it(src_file, tst_file)
+    #     if msg:
+    #         self.fail(tst_file + ': ' + msg)
+    #===========================================================================
+
+    def test_ged(self):
+        """ Run a Gedcom export test """
+        config.set('preferences.place-auto', True)
+        src_file = 'exp_sample.gramps'
+        tst_file = 'exp_sample_ged.ged'
+        msg = do_it(src_file, tst_file, gedfilt)
```

Comment:
If `do_it` were a method of this class, then it could call `self.fail` directly, which it seems needs to be done every time.